scipy/signal/ltisys.py

```diff
@@ -500,6 +501,419 @@ def freqresp(self, w=None, n=10000):
         return freqresp(self, w=w, n=n)
 
 
+class TransferFunction(lti):
+    """Linear Time Invariant system class in transfer function form.
+
+    Parameters
+    ----------
+    system : arguments
+        The `TransferFunction` class can be instantiated with 1 or 2 arguments.
+        The following gives the number of input arguments and their
+        interpretation:
+
+            * 1: (lti system: StateSpace, TransferFunction or ZerosPolesGain)
+            * 2: (numerator, denominator)
+
+    """
+    def __new__(cls, *system):
+        """Handle object conversion if input is an instance of lti"""
+        if len(system) == 1:
+            if isinstance(system[0], TransferFunction):
+                return copy.deepcopy(system[0])
+            if isinstance(system[0], lti):
+                return system[0].to_tf()
+
+        # No special conversion needed
+        return super(TransferFunction, cls).__new__(cls)
+
+    def __init__(self, *system):
+        """Initialize the state space LTI system
+
+        Parameters
+        ----------
+        system : arguments
+            The following arguments are possible
+            * an instance of the lti class (StateSpace, TransferFunction
+              or ZerosPolesGain)
+            * (numerator, denominator)
```

Comment:
Have you checked how this looks in docs? This might need to be indented to render properly